src/api/system.c

```diff
@@ -769,7 +769,7 @@ static int f_load_native_plugin(lua_State *L) {
   const char *path = luaL_checkstring(L, 2);
   void *library = SDL_LoadObject(path);
   if (!library)
-    return luaL_error(L, "Unable to load %s: %s", name, SDL_GetError());
+    return (lua_pushstring(L, SDL_GetError()), lua_error(L));
```

Comment:
This doesn't look correct.

Wouldn't something like this make more sense?
```C
return luaL_error(L, "%s", SDL_GetError());
```

Reply:
Not really, `lua_error()` takes whatever that's on the stack and throws an error with it. `luaL_error()` is just `lua_pushfstring` and `lua_error` packaged together, but in this case there's no need to call `lua_pushfstring`.

Reply:
My main complaint is the `return (a(), b());`.